torch/optim/optimizer.py

```diff
@@ -132,6 +132,11 @@ def _view_as_real(params, *state_and_grads):
             for s in state_and_grads:
                 s[i] = torch.view_as_real(s[i])
 
+def _get_scalar_dtype(is_fused=None):
+    if is_fused:
+        return torch.float32
+    return torch.float64 if torch.get_default_dtype() == torch.float64 else torch.float32
```

Comment:
I don't remember the story for default dtype vs default Tensor type. Do we need to check both here?

Reply:
I don't think we do, get_default_dtype seems to work as expected:
<img width="764" alt="image" src="https://github.com/pytorch/pytorch/assets/31798555/be395f5a-69f4-495c-8944-e49b234d8dde">